cmd/notation/list.go

```diff
@@ -1,43 +1,51 @@
 package main
 
 import (
-	"errors"
 	"fmt"
 
-	"github.com/urfave/cli/v2"
+	"github.com/spf13/cobra"
 )
 
-var listCommand = &cli.Command{
-	Name:      "list",
-	Usage:     "List signatures from remote",
-	Aliases:   []string{"ls"},
-	ArgsUsage: "<reference>",
-	Flags: []cli.Flag{
-		flagUsername,
-		flagPassword,
-	},
-	Action: runList,
+type listOpts struct {
+	SecureFlagOpts
+	reference string
 }
 
-func runList(ctx *cli.Context) error {
-	// initialize
-	if !ctx.Args().Present() {
-		return errors.New("no reference specified")
+func listCommand(opts *listOpts) *cobra.Command {
+	if opts == nil {
+		opts = &listOpts{}
+	}
+	cmd := &cobra.Command{
+		Use:     "list [reference]",
+		Aliases: []string{"ls"},
+		Short:   "List signatures from remote",
+		Args:    cobra.ExactArgs(1),
```

Comment:
Same here. I feel like cobra's ExactArgs error "Error: accepts 1 arg(s), received 0" is kind of vague. The original logic checks ctx.Args().Present(), if not, then it returns error "no reference specified", which explicitly tells the user what is missing.

Reply:
@binbin-li  @qweeah , Do you prefer using `cobra.ExactArgs` or handling args in `PreRunE`?

Reply:
@chloeyin We should customize a arg validation function and provide to the `Args`.
@patrickzheng200 Good catch. I should also apply this to ORAS.